src/error.rs

```diff
@@ -0,0 +1,57 @@
+/// Crypto error variants
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum Error {
+    /// Invalid secp256k1 secret key
+    InvalidSecretKey,
+
+    /// Invalid secp256k1 public key
+    InvalidPublicKey,
+
+    /// Invalid secp256k1 signature message
+    InvalidMessage,
+
+    /// Invalid secp256k1 signature
+    InvalidSignature,
+
+    /// Out of preallocated memory
+    NotEnoughMemory,
+}
+
+#[cfg(feature = "std")]
+mod use_std {
```

Comment:
It is true, the library is separated via feature and we have a test for the CI to guarantee it will build in `no-std` without `secp256k1`.

We can represent the types in `no-std`, but we can't sign. If we ever need to sign under `no-std`, then we need to consider using other libs.